src/worldcereal/utils/legend.py

```diff
@@ -0,0 +1,134 @@
+import json
+import os
+import subprocess
+from pathlib import Path
+
+from loguru import logger
+
+ARTIFACTORY_BASE_URL = (
+    "https://artifactory.vgt.vito.be/artifactory/auxdata-public/worldcereal/"
+)
+
+
+def _get_artifactory_credentials():
+    """Get credentials for upload and delete operations on Artifactory.
+    Returns
+    -------
+    tuple (str, str)
+        Tuple containing the Artifactory username and password.
+    Raises
+    ------
+    ValueError
+        if ARTIFACTORY_USERNAME or ARTIFACTORY_PASSWORD are not set as environment variables.
+    """
+
+    artifactory_username = os.getenv("ARTIFACTORY_USERNAME")
+    artifactory_password = os.getenv("ARTIFACTORY_PASSWORD")
+
+    if not artifactory_username or not artifactory_password:
+        raise ValueError(
+            "Artifactory credentials not found. "
+            "Please set ARTIFACTORY_USERNAME and ARTIFACTORY_PASSWORD environment variables."
+        )
+
+    return artifactory_username, artifactory_password
+
+
+def upload_legend_csv_artifactory(
+    srcpath: Path,
+    date: str,
+) -> str:
+    """Uploads a CSV file containing the worldcereal land cover/crop type legend to Artifactory.
+    Parameters
+    ----------
+    srcpath : Path
+        Path to csv file that needs to be uploaded to Artifactory.
+    date : str
+        Date tag to be added to the filename. Should be in format YYYYMMDD.
+    Returns
+    -------
+    str
+        artifactory download link
+    Raises
+    ------
+    FileNotFoundError
+        if srcpath does not exist
+    """
+    if not srcpath.is_file():
+        raise FileNotFoundError(f"Required file `{srcpath}` not found.")
+
+    # Get Artifactory credentials
+    artifactory_username, artifactory_password = _get_artifactory_credentials()
+
+    # We  upload the file with a specific date tag and also with a "latest" tag
+    target_names = [
+        f"WorldCereal_LC_CT_legend_{date}.csv",
+        "WorldCereal_LC_CT_legend_latest.csv",
+    ]
+    targetpaths = [f"{ARTIFACTORY_BASE_URL}legend/{n}" for n in target_names]
+
+    for targetpath in targetpaths:
+        logger.info(f"Uploading `{srcpath}` to `{targetpath}`")
+
+        cmd = (
+            f"curl -u{artifactory_username}:{artifactory_password} -T {srcpath} "
+            f'"{targetpath}"'
+        )
+
+        output, _ = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, shell=True
+        ).communicate()
+        decoded_output = output.decode("utf-8")
+
+        # Parse as JSON if applicable
+        try:
+            parsed_output = json.loads(decoded_output)
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Failed to parse output as JSON: {decoded_output}") from e
+
+    # Access the desired value
+    return parsed_output.get("downloadUri")
+
+
+def download_latest_legend_from_artifactory(download_path: Path) -> Path:
```

Comment:
not sure yet how it's used, but it could be useful if we only want to access the contents, that you allow the option to download to temporary file that is automatically destroyed after the contents have been loaded. Because what do you want to happen ultimately? Do you want to get the contents of the CSV? THen why not immediately returning that instead of the path to the file?